sql/src/main/java/io/crate/jobs/UpsertByIdContext.java

```diff
@@ -113,6 +119,7 @@ public void close() {
 
     @Override
     public void kill() {
-        throw new UnsupportedOperationException("kill is not implemented");
+        futureResult.setException(new JobKilledException());
```

Comment:
it could be that the futureResult is set twice as the close call happens after setting it here. Could that be a problem?